app/src/main/res/layout-sw600dp-land/onboarding_app_language_selection_fragment.xml

```diff
@@ -0,0 +1,163 @@
+      android:text="@string/onboarding_language_activity_subtitle"
+      android:textSize="@dimen/onboarding_shared_text_size_large"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toBottomOf="@id/onboarding_language_title" />
+
+    <TextView
+      android:id="@+id/onboarding_language_text"
+      style="@style/OnboardingLanguageMessageStyle"
+      android:layout_marginTop="@dimen/tablet_shared_margin_x_small"
+      android:text="@string/onboarding_language_activity_text"
+      android:textSize="@dimen/onboarding_shared_text_size_medium"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toBottomOf="@id/onboarding_language_subtitle" />
+
+    <androidx.constraintlayout.widget.Guideline
+      android:id="@+id/onboarding_language_image_guide"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:orientation="horizontal"
+      app:layout_constraintGuide_percent="0.34" />
+
+    <androidx.constraintlayout.widget.Guideline
+      android:id="@+id/onboarding_language_center_guide"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:orientation="horizontal"
+      app:layout_constraintGuide_percent="0.45" />
+
+    <org.oppia.android.app.customview.OppiaCurveBackgroundView
+      android:id="@+id/onboarding_app_language_background"
+      android:layout_width="match_parent"
+      android:layout_height="0dp"
+      app:customBackgroundColor="@color/component_color_onboarding_shared_white_color"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintTop_toTopOf="@id/onboarding_language_center_guide" />
+
+    <ImageView
+      android:id="@+id/onboarding_app_language_image"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:contentDescription="@string/onboarding_otter_content_description"
+      android:scaleType="centerCrop"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toBottomOf="@id/onboarding_language_image_guide"
+      app:srcCompat="@drawable/otter" />
+
+    <TextView
+      android:id="@+id/onboarding_language_label"
+      style="@style/OnboardingLanguageLabelStyle"
+      android:layout_marginTop="@dimen/tablet_shared_margin_large"
+      android:text="@string/onboarding_language_activity_select_label"
+      app:layout_constraintBottom_toTopOf="@id/onboarding_language_dropdown_background"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toBottomOf="@id/onboarding_app_language_image"
+      app:layout_constraintVertical_chainStyle="packed" />
+
+    <RelativeLayout
```

Comment:
I chose RelativeLayout as a way to group the views. By **flat ConstraintLayout**, do you mean to avoid child ViewGroups i.e. not to replace RelativeLayout with ConstraintLayout, but to instead arrange the spinner and icons within the parent view?

Reply:
Fixed this using MaterialCardview and Material TextInputLayout.

Reply:
Ah sorry, I did mean a single ConstraintLayout. FWIW the latest changes look really good!